### PR TITLE
test: add benchmark for all hash functions Sum

### PR DESCRIPTION
**Pull request by Jorropo**

### CODE REVIEW

File: sum_test.go [hunk 1]
```diff
@@ -177,3 +180,46 @@ func TestBasicSum(t *testing.T) {
 		}
 	}
 }
+
+var sink []byte
```
Comment (marten-seemann) [inline sum_test.go:184]: I think this has to be an exported type, if you want to disable certain compiler optimizations.

File: sum_test.go [hunk 2]
```diff
@@ -177,3 +180,46 @@ func TestBasicSum(t *testing.T) {
...
+
+type codeNamePair struct {
+	id   uint64
+	name string
+}
+
+func BenchmarkSumAllLarge(b *testing.B) {
+	var data [1024 * 1024 * 16]byte
+	_, err := io.ReadFull(rand.Reader, data[:])
```
Comment (marten-seemann) [inline sum_test.go:193]: `rand.Read` does the same thing. We probably shouldn't use cryptographic random here though, getting 16 MB from there might be slow. Better to use a `math/rand` reader.

File: sum_test.go (hunk 1 above)
Comment (Jorropo) [reply sum_test.go:184]: Your point make sense, but I don't think the compiler is that smart yet.

Comment (marten-seemann) [reply sum_test.go:184]: I think it is. At least I've read that advice in some benchmarking tutorial (can't find the link though).

Comment (Jorropo) [reply sum_test.go]: Fixed

File: sum_test.go (hunk 2 above)
Comment (Jorropo) [reply sum_test.go]: Fixed

**Review by marten-seemann — APPROVED** [review]

**Pull request merged**